plugins/module_utils/openssh/cryptography_openssh.py

```diff
@@ -0,0 +1,604 @@
+        """
+
+        comment = extract_comment(path)
+        asym_keypair = Asymmetric_Keypair.load(path, passphrase, 'SSH')
+        openssh_privatekey = cls.encode_openssh_privatekey(asym_keypair)
+        openssh_publickey = cls.encode_openssh_publickey(asym_keypair, comment)
+        fingerprint = calculate_fingerprint(openssh_publickey)
+
+        return cls(
+            asym_keypair=asym_keypair,
+            openssh_privatekey=openssh_privatekey,
+            openssh_publickey=openssh_publickey,
+            fingerprint=fingerprint,
+            comment=comment
+        )
+
+    @classmethod
+    def encode_openssh_privatekey(cls, asym_keypair):
+        """Returns an OpenSSH encoded private key for a given keypair
+
+           :asym_keypair: Asymmetric_Keypair from the private key is extracted
+        """
+
+        # OpenSSH formatted private keys are not available in Cryptography <3.0
+        try:
+            privatekey_format = serialization.PrivateFormat.OpenSSH
+        except AttributeError:
+            privatekey_format = serialization.PrivateFormat.PKCS8
+
+        encoded_privatekey = asym_keypair.private_key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=privatekey_format,
+            encryption_algorithm=asym_keypair.encryption_algorithm
+        )
+
+        return encoded_privatekey
+
+    @classmethod
+    def encode_openssh_publickey(cls, asym_keypair, comment):
+        """Returns an OpenSSH encoded public key for a given keypair
+
+           :asym_keypair: Asymmetric_Keypair from the public key is extracted
+           :comment: Comment to apply to the end of the returned OpenSSH encoded public key
+        """
+        encoded_publickey = asym_keypair.public_key.public_bytes(
+            encoding=serialization.Encoding.OpenSSH,
+            format=serialization.PublicFormat.OpenSSH,
+        )
+
+        validate_comment(comment)
+
+        encoded_publickey += (" %s" % comment).encode(encoding=_TEXT_ENCODING)
+
+        return encoded_publickey
+
+    def __init__(self, asym_keypair, openssh_privatekey, openssh_publickey, fingerprint, comment):
+        """
+           :asym_keypair: An Asymmetric_Keypair object from which the OpenSSH encoded keypair is derived
+           :openssh_privatekey: An OpenSSH encoded private key
+           :openssh_privatekey: An OpenSSH encoded public key
+           :fingerprint: The fingerprint of the OpenSSH encoded public key of this keypair
+           :comment: Comment applied to the OpenSSH public key of this keypair
+        """
+
+        self.__asym_keypair = asym_keypair
+        self.__openssh_privatekey = openssh_privatekey
+        self.__openssh_publickey = openssh_publickey
+        self.__fingerprint = fingerprint
+        self.__comment = comment
+
+    def __eq__(self, other):
+        return self.fingerprint == other.fingerprint and self.comment == other.comment
```

Comment:
I would rather compare the keys themselves, instead of only comparing the fingerprint. Or are some of the key objects not comparable?

(Also this ignores the passphrase.)